### DISCUSSION TITLE
[Feature]: DataTable as Input

### DISCUSSION THREAD

**Discussion by hughess** [evidence-dev/evidence#2433] [General]
### Description

DataTable ability to act as an interactive input - click a row to select the values from that row and populate an input variable.

The input variable could then be referenced by column. E.g., if a table had `name=myinput` and columns `category`, `region`, and `sales`, you could reference it with any of:
- `inputs.myinput.category`
- `inputs.myinput.region`
- `inputs.myinput.sales`

**Comment by richard-lennox** [comment]
Would be particularly useful where the target is one of the Map components. This would require the Map components to accept a target value, and then zoom to and highlight the selected polygon. 

This covers the use-case where you might have a fairly complex geospatial representation of the DataTable (with say 100+ rows) in a Map, but the user doesn't necessarily know where on the Map the data rows are. They can search/find the appropriate information in the DataTable, click on the row, and then this updates/selects the right data in the Map. A reverse of the current use of the Map component as an interactive input.

**Comment by jaroet** [comment]
I have a situation where my dropdown contains about 360 items. This is hard to navigate using the current dropdown UI. 
A solution like described above would be nice ... but I can imagine an improvement of the dropdown component might also be an option. 

for instance an option to set a `dropdownUI=table`. Then the source query will be shown as a table showing all available fields in a table and an option to define the number of rows among others and also using the full width of the project. Maybe even have all options that are currently in datatable (although that might  be a little much).
